compiler/runtime/TRRelocationRecord.hpp

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
```

Comment:
I think this file should be named `RelocationRecord.hpp` for it to fit in with the extensible classes model.

Reply:
This is named TRRelocationRecord to avoid breaking the build of the cmake, because otherwise it would shadow / pick up the wrong file. Thank you for the question!

Reply:
I'm not sure I follow. The way extensible classes works is if you have `OMRRelocationRecord.hpp` which has the `OMR::RelocationRecord` class. Then in `RelocationRecord.hpp`, you have the `TR::RelocationRecord` class which extends `OMR::RelocationRecord`.

The whole point is that the make system would find the right files because everywhere you're only including `RelocationRecord.hpp`. If say openj9 wanted to extend, it would have to create a file `J9RelocationRecord` to hold `J9::RelocationRecord` which extends `OMR::RelocationRecord` and then a `RelocationRecord.hpp` file which holds `TR::RelocationRecord` which extends `J9::RelocationRecord.

Reply:
The problem here is in the fact there are two build systems that are currently used. The autoconf and cmake one. Relocation records in openj9 are not extensible, and cmake will pick them up and overwrite the OMR version if the names will match, because it seems to include whole contents directory, which will result in undefined symbols. In autoconf case, we can chose which ones to pick up, and if we configure it to pick up only what we want it solves the problem, but that would require a simultaneous pr into OpenJ9 on modification of the buildsystem or making Openj9 RelocationRecords extensible, and  that's a breaking change. Since both build systems are required to be supported, I've came up with the following solution (temporary) until the work on making Openj9 relocation records extensible begins. Does this make things clearer?

Reply:
The change to OpenJ9 would include renaming OpenJ9 RelocationRecords into J9RelocationRecord. After using them, or choosing one build system, the file can be renamed back.

Reply:
Ah ok I see, fair enough

Reply:
Thank you for the comment. The list of changes needs to be created and is not detailed as of now.

Reply:
looked like #5469 has been created now so recording it here. Thanks